src/sound/WAVFile.as

```diff
@@ -60,14 +60,22 @@ public class WAVFile {
 		// read format chunk
 		var formatChunk:ByteArray = extractChunk('fmt ', waveData);
 		if (formatChunk.length < 16) throw Error("WAVFile: format chunk is too small");
-		var encoding:int = formatChunk.readShort();
+		var encoding:uint = formatChunk.readUnsignedShort();
 
 		result.encoding = encoding;
-		result.channels = formatChunk.readShort();
-		result.samplesPerSecond = formatChunk.readInt();
-		result.bytesPerSecond = formatChunk.readInt();
-		result.blockAlignment = formatChunk.readShort();
-		result.bitsPerSample = formatChunk.readShort();
+		result.channels = formatChunk.readUnsignedShort();
+		result.samplesPerSecond = formatChunk.readUnsignedInt();
+		result.bytesPerSecond = formatChunk.readUnsignedInt();
+		result.blockAlignment = formatChunk.readUnsignedShort();
+		result.bitsPerSample = formatChunk.readUnsignedShort();
+		if (formatChunk.length >= 18 && encoding == 0xFFFE) {
+			var extensionSize:uint = formatChunk.readUnsignedShort();
+			if (extensionSize == 22) {
+				result.validBitsPerSample = formatChunk.readUnsignedShort();
+				result.channelMask = formatChunk.readUnsignedInt();
+				result.encoding = encoding = formatChunk.readUnsignedShort();
```

Comment:
Have you tested this `encoding` assignment specifically? Based on the "Extensible Format" section of [this page](http://www-mmsp.ece.mcgill.ca/documents/audioformats/wave/wave.html) it looks to me like you're reading part of a sub-format GUID into `encoding`. Do your docs disagree about what comes after `dwChannelMask`?